tests/test_basic_ops_dict.py

```diff
@@ -398,10 +403,29 @@ def test_dict_pop_error(cfg: Dict[Any, Any], key: Any, expectation: Any) -> None
             "incompatible_key_type",
             False,
         ),
+        (  # unexpected failure?
+            DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str),
+            "FOO",
+            False,
+        ),
```

Comment:
I was surprised that this test fails.
This was surprising to me because two of the previous tests asserts successfully that
```
assert "FOO" not in OmegaConf.create({Enum1.FOO: 1, "b": {}})
assert "incompatible_key_type" not in OmegaConf.create(DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str))
```
whereas this test fails when we try
```
assert "FOO" not in OmegaConf.create(DictConfig(content={Enum1.FOO: "foo"}, key_type=Enum1, element_type=str))
```

This means that the string `"FOO"` is being coerced to the `Enum1` type by the `DictConfig.__contains__` method, which seems to contradict the use of the string `"incompatible_key_type"` in the previous test (which suggests that the intended behavior is for `__contains__(self, key)` is to return `False` when `self` has `key_type` that is different from the type of the passed `key`).

Reply:
```
>>> "FOO" in omegaconf.DictConfig({Enum1.FOO: "foo"})
False
>>> "FOO" in omegaconf.DictConfig({Enum1.FOO: "foo"}, key_type=Enum1)
True
```
@omry is this the expected behavior?